frontends/concrete-python/concrete/fhe/mlir/context.py

```diff
@@ -520,10 +520,12 @@ def bitwise(
                 )
             self.error(highlights)
 
-        assert self.is_bit_width_compatible(resulting_type, x, y)
+        assert self.is_bit_width_compatible(x, y)
         assert resulting_type.is_encrypted and x.is_encrypted and y.is_encrypted
 
-        if x.original_bit_width + y.original_bit_width <= resulting_type.bit_width:
+        internal_result_type = max(x.type, y.type, key=lambda t: len(t.shape))
```

Comment:
This is wrong, input shapes can be `(3, 1, 2)` and `(4, 2)` which would make the result `(3, 4, 2)` and this behavior is not captured here.

Reply:
(you can use https://numpy.org/doc/stable/reference/generated/numpy.broadcast_shapes.html to do this easily)